File: app/src/main/java/de/xikolo/receivers/NetworkChangeReceiver.java
package de.xikolo.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.net.ConnectivityManager;

import org.greenrobot.eventbus.EventBus;

import de.xikolo.events.NetworkStateEvent;
import de.xikolo.utils.NetworkUtil;

public class NetworkChangeReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(final Context context, final Intent intent) {
        if (ConnectivityManager.CONNECTIVITY_ACTION.equals(intent.getAction())) {
            boolean isOnline = NetworkUtil.isOnline();
            EventBus.getDefault().postSticky(new NetworkStateEvent(isOnline));
        }
    }

}

File: app/src/main/java/de/xikolo/receivers/NetworkChangeReceiver.kt
package de.xikolo.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.net.ConnectivityManager
import de.xikolo.App
import de.xikolo.utils.NetworkUtil

class NetworkChangeReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (ConnectivityManager.CONNECTIVITY_ACTION == intent.action) {
            val isOnline = NetworkUtil.isOnline()

            if (isOnline) {
                App.instance.state.connectivity.online()
            } else {
                App.instance.state.connectivity.offline()
            }
        }
    }

}
